### PR TITLE
Exports off of couch - redux

### PR DESCRIPTION
**Pull request by NoahCarnahan**
This PR un-reverts https://github.com/dimagi/commcare-hq/pull/10477, and continues on that work.
- Fixes a variety of bugs
- Implements missing views
- Adds missing export row number functionality
- Adds a `user_type` field to the xform es index

This PR is not ready to merge, just putting it up now for review. I think I want to run QA on staging, and we also need to time the xform reindex appropriately. I also need to do a few more things, like actually use the `user_type` field in the export filters.

Should be reviewed commit-by-commit. The first commit was PR https://github.com/dimagi/commcare-hq/pull/10477, so all those changes were reviewed there.

### CODE REVIEW

**Comment by dimagimon** [comment]
There are 7 errors:

* corehq/apps/export/models/new.py, line 255 - E501 line too long (131 > 115 characters)
* corehq/apps/export/tests/test_daily_saved_exports.py, line 7 - E302 expected 2 blank lines, found 1
* corehq/apps/export/tests/test_daily_saved_exports.py, line 29 - E301 expected 1 blank line, found 0
* corehq/apps/export/views.py, line 1557 - E303 too many blank lines (3)
* corehq/apps/reports/urls.py, line 88 - E501 line too long (142 > 115 characters)
* corehq/pillows/mappings/xform_mapping.py, line 6 - E303 too many blank lines (3)
* corehq/couchapps/export_instances_by_is_daily_saved/views/view/map.js, line 1 - Missing name in function declaration.

File: corehq/apps/export/models/new.py [hunk 1]
```diff
@@ -222,14 +241,18 @@ def get_column(self, item_path):
...
+    def _get_sub_documents_helper(path, row_docs):
         """
         Return each instance of a repeat group at the path from the given docs.
         If path is [], just return the docs
 
-        >>> TableConfiguration()._get_sub_documents(['foo'], [{'foo': {'bar': 'a'}}, {'foo': {'bar': 'b'}}])
+        >>> TableConfiguration._get_sub_documents_helper(['foo'], [{'foo': {'bar': 'a'}}, {'foo': {'bar': 'b'}}])
         [{'bar': 'a'}, {'bar': 'b'}]
-        >>> TableConfiguration()._get_sub_documents(['foo', 'bar'], [{'foo': [{'bar': {'baz': 'a'}}, {'bar': {'baz': 'b'}},]}]
+        >>> TableConfiguration._get_sub_documents_helper(['foo', 'bar'], [{'foo': [{'bar': {'baz': 'a'}}, {'bar': {'baz': 'b'}},]}]
```
Comment (dimagimon) [inline corehq/apps/export/models/new.py]: E501 line too long (131 > 115 characters)

File: corehq/apps/reports/urls.py [hunk 2]
```diff
@@ -85,6 +85,7 @@
     url(r"^export/bulk/download/$", "export_default_or_custom_data", name="export_bulk_download", kwargs=dict(bulk_export=True)),
     # saved
     url(r"^export/saved/download/(?P<export_id>[\w\-]+)/$", "hq_download_saved_export", name="hq_download_saved_export"),
+    url(r"^export/new/saved/download/(?P<export_instance_id>[\w\-]+)/$", "hq_download_new_saved_export", name="hq_download_new_saved_export"),
```
Comment (dimagimon) [inline corehq/apps/reports/urls.py]: E501 line too long (142 > 115 characters)

File: corehq/couchapps/export_instances_by_is_daily_saved/views/view/map.js [hunk 3]
```diff
@@ -0,0 +1,5 @@
+function(doc) {
```
Comment (dimagimon) [inline corehq/couchapps/export_instances_by_is_daily_saved/views/view/map.js]: Missing name in function declaration.

**Comment by benrudolph** [comment]
sweet this is looking real nice

File: corehq/apps/export/models/new.py (hunk 1 above)
Comment (dimagimon) [inline corehq/apps/export/models/new.py]: E501 line too long (131 > 115 characters)

File: corehq/couchapps/export_instances_by_is_daily_saved/views/view/map.js (hunk 3 above)
Comment (dimagimon) [inline corehq/couchapps/export_instances_by_is_daily_saved/views/view/map.js]: Missing name in function declaration.

**Comment by snopoke** [comment]
not mergable

File: corehq/pillows/case.py [hunk 4]
```diff
@@ -104,3 +112,21 @@ def get_couch_case_reindexer():
 
 def get_sql_case_reindexer():
     return PillowReindexer(get_sql_case_to_elasticsearch_pillow(), SqlCaseChangeProvider())
+
+@quickcache(['domain', 'user_id'], timeout=ONE_DAY)
```
Comment (dimagimon) [inline corehq/pillows/case.py]: E302 expected 2 blank lines, found 1

File: corehq/pillows/case.py [hunk 5]
```diff
@@ -104,3 +112,21 @@ def get_couch_case_reindexer():
...
+def _get_username(domain, user_id):
+    """
+    Get the username for the given user_id. We are replicating the beahvior of
+    corehq.apps.reports.util.get_all_users_by_domain
+    """
+    # TODO: This assumes that the form index will be built before the case index. Will that always be true? Is this safe?
```
Comment (dimagimon) [inline corehq/pillows/case.py]: E501 line too long (121 > 115 characters)

File: corehq/couchapps/export_instances_by_is_daily_saved/views/view/map.js (hunk 3 above)
Comment (dimagimon) [inline corehq/couchapps/export_instances_by_is_daily_saved/views/view/map.js]: Missing name in function declaration.

File: corehq/pillows/utils.py [hunk 6]
```diff
@@ -1,4 +1,23 @@
+from corehq.apps.commtrack.const import COMMTRACK_USERNAME
+from corehq.apps.reports.models import HQUserType
```
Comment (dimagimon) [inline corehq/pillows/utils.py]: F401 'HQUserType' imported but unused

File: corehq/couchapps/export_instances_by_is_daily_saved/views/view/map.js (hunk 3 above)
Comment (dimagimon) [inline corehq/couchapps/export_instances_by_is_daily_saved/views/view/map.js]: Missing name in function declaration.

Comment (dimagimon) [inline corehq/couchapps/export_instances_by_is_daily_saved/views/view/map.js]: Missing name in function declaration.

**Comment by NoahCarnahan** [comment]
@dimagi/scale-team this is ready for review again.
The primary changes since opening the PR are:
- Add user type to form and case es indexes
- eliminate the "Admin" export filter
- track download progress


I think we're very close to being able to deploy. The two things left on my radar are:
- Implement the special system properties
- Finish off the functions that migrate existing exports

My plan is to deploy this branch to staging if the travis build passes

**Comment by benrudolph** [comment]
this looks good to me :+1: ping when you think you've done enough testing on staging. would be good to and do the other two things on your radar in a separate PR

File: corehq/couchapps/export_instances_by_is_daily_saved/views/view/map.js (hunk 3 above)
Comment (dimagimon) [inline corehq/couchapps/export_instances_by_is_daily_saved/views/view/map.js]: Missing name in function declaration.

Comment (dimagimon) [inline corehq/couchapps/export_instances_by_is_daily_saved/views/view/map.js]: Missing name in function declaration.

**Comment by benrudolph** [comment]
dang we were supposed to merge this on friday weren't we

**Comment by NoahCarnahan** [comment]
Yeah, sorry about that. Let's discuss the best approach when you come in.

File: corehq/couchapps/export_instances_by_is_daily_saved/views/view/map.js (hunk 3 above)
Comment (dimagimon) [inline corehq/couchapps/export_instances_by_is_daily_saved/views/view/map.js]: Missing name in function declaration.

File: corehq/apps/export/tests/test_export_filters.py [hunk 7]
```diff
@@ -11,15 +11,26 @@
...
     IsClosedFilter,
     OwnerFilter,
-    OR)
+    OR,
+    FormSubmittedByFilter,
+)
 from corehq.apps.export.models.new import (
     CaseExportInstance,
+    FormExportInstance)
+from corehq.apps.export.tests.util import (
```
Comment (dimagimon) [inline corehq/apps/export/tests/test_export_filters.py]: F401 'DEFAULT_APP_ID' imported but unused

Comment (dimagimon) [inline corehq/apps/export/tests/test_export_filters.py]: F401 'DEFAULT_XMLNS' imported but unused

File: corehq/couchapps/export_instances_by_is_daily_saved/views/view/map.js (hunk 3 above)
Comment (dimagimon) [inline corehq/couchapps/export_instances_by_is_daily_saved/views/view/map.js]: Missing name in function declaration.

**Comment by NoahCarnahan** [comment]
@benrudolph Alright I've removed the es index changes from this PR, so it should be good to merge now. I left in the new functions like `FormES.user_type()`, which require reindex to work for simplicities sake, but happy to remove them if you think it could cause confusion to keep them.

**Comment by benrudolph** [comment]
yeeha :horse:! nope i think it's good to leave

**Comment by benrudolph** [comment]
:fireworks: :fireworks: :fireworks:

**Comment by czue** [comment]
:balloon:

**Comment by NoahCarnahan** [comment]
:joy:

**Pull request merged**